sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java

```diff
@@ -0,0 +1,1069 @@
+        return leaveGroup(group, nextAckId());
+    }
+
+    /**
+     * Leaves a group.
+     *
+     * @param group the group name.
+     * @param ackId the ackId. Client will provide auto increment ID, if set to {@code null}.
+     * @return the result.
+     */
+    public Mono<WebPubSubResult> leaveGroup(String group, Long ackId) {
+        return sendMessage(new LeaveGroupMessage().setGroup(group).setAckId(ackId))
+            .then(waitForAckMessage(ackId)).retryWhen(sendMessageRetrySpec)
+            .map(result -> {
+                groups.compute(group, (k, v) -> {
+                    if (v == null) {
+                        return new WebPubSubGroup(group).setJoined(false);
+                    } else {
+                        return v.setJoined(false);
+                    }
+                });
+                return result;
+            });
+    }
+
+    /**
+     * Sends message to group.
+     *
+     * @param group the group name.
+     * @param content the data as WebPubSubDataType.TEXT.
+     * @return the result.
+     */
+    public Mono<WebPubSubResult> sendToGroup(String group, String content) {
+        return sendToGroup(group, BinaryData.fromString(content), WebPubSubDataType.TEXT);
+    }
+
+    /**
+     * Sends message to group.
+     *
+     * @param group the group name.
+     * @param content the data as WebPubSubDataType.TEXT.
+     * @param options the options.
+     * @return the result.
+     */
+    public Mono<WebPubSubResult> sendToGroup(String group, String content, SendToGroupOptions options) {
+        return sendToGroup(group, BinaryData.fromString(content), WebPubSubDataType.TEXT, options);
+    }
+
+    /**
+     * Sends message to group.
+     *
+     * @param group the group name.
+     * @param content the data.
+     * @param dataType the data type.
+     * @return the result.
+     */
+    public Mono<WebPubSubResult> sendToGroup(String group, BinaryData content, WebPubSubDataType dataType) {
+        return sendToGroup(group, content, dataType, new SendToGroupOptions().setAckId(nextAckId()));
+    }
+
+    /**
+     * Sends message to group.
+     *
+     * @param group the group name.
+     * @param content the data.
+     * @param dataType the data type.
+     * @param options the options.
+     * @return the result.
+     */
+    public Mono<WebPubSubResult> sendToGroup(String group, BinaryData content, WebPubSubDataType dataType,
+                                             SendToGroupOptions options) {
+        Objects.requireNonNull(group);
+        Objects.requireNonNull(content);
+        Objects.requireNonNull(dataType);
+        Objects.requireNonNull(options);
+
+        long ackId = options.getAckId() != null ? options.getAckId() : nextAckId();
+
+        Object data = content;
+        if (dataType == WebPubSubDataType.BINARY || dataType == WebPubSubDataType.PROTOBUF) {
+            data = Base64.getEncoder().encodeToString(content.toBytes());
```

Comment:
A bit concern here. `WebPubSubClient` should keep this content as is and leave it to encoder (json/protobuf/..) to encode. Can be improved in future

Reply:
Got your point, this should be done in WebPubSubProtocol subclass.

Reply:
At present, moved the logic to Encoder (that would belong to a specific Protocol class).